--- /dev/null
+++ b/cmake/FindAnalitza.cmake
@@ -0,0 +1,27 @@
+# Find libanalitza
+# Once done this will define
+#
+#  ANALITZA_FOUND    - system has Analitza Library
+#  ANALITZA_INCLUDES - the Analitza include directory
+#  ANALITZA_LIBS     - link these to use Analitza
+#  ANALITZA_VERSION  - the version of the Analitza Library
+
+# Copyright (c) 2011, Aleix Pol Gonzalez <aleixpol@kde.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+find_library(ANALITZA_LIBRARY NAMES analitza HINTS ${KDE4_LIB_INSTALL_DIR} ${QT_LIBRARY_DIR})
+find_library(ANALITZAGUI_LIBRARY NAMES analitzagui HINTS ${KDE4_LIB_INSTALL_DIR} ${QT_LIBRARY_DIR})
+
+find_path(ANALITZA_INCLUDE_DIR NAMES analitza/analitzaexport.h HINTS ${KDE4_INCLUDE_INSTALL_DIR} ${QT_INCLUDE_DIR} ${INCLUDE_INSTALL_DIR})
+
+if(ANALITZA_INCLUDE_DIR AND ANALITZA_LIBRARY)
+   set(ANALITZA_LIBS ${analitza_LIB_DEPENDS} ${ANALITZA_LIBRARY} ${ANALITZAGUI_LIBRARY})
+   set(ANALITZA_INCLUDES ${ANALITZA_INCLUDE_DIR}/KDE ${ANALITZA_INCLUDE_DIR})
+endif(ANALITZA_INCLUDE_DIR AND ANALITZA_LIBRARY)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Analitza  DEFAULT_MSG  ANALITZA_INCLUDE_DIR ANALITZA_LIBRARY)
+
+mark_as_advanced(ANALITZA_INCLUDE_DIR ANALITZA_LIBRARY)
--- a/src/backends/kalgebra/CMakeLists.txt
+++ b/src/backends/kalgebra/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories(${analitza_SOURCE_DIR}/../)
+include_directories(${ANALITZA_INCLUDES})
 
 set( KAlgebraBackend_SRCS
  kalgebrasyntaxhelpobject.cpp 
@@ -10,7 +10,7 @@ set( KAlgebraBackend_SRCS
 )
 
 kde4_add_plugin( cantor_kalgebrabackend ${KAlgebraBackend_SRCS} )
-target_link_libraries( cantor_kalgebrabackend ${KDE4_KDEUI_LIBS} cantorlibs analitza analitzagui)
+target_link_libraries( cantor_kalgebrabackend ${KDE4_KDEUI_LIBS} cantorlibs ${ANALITZA_LIBS})
 
 install( FILES kalgebrabackend.desktop  DESTINATION ${SERVICES_INSTALL_DIR}/cantor)
 install(TARGETS cantor_kalgebrabackend DESTINATION ${PLUGIN_INSTALL_DIR})
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,9 @@ include(KDE4Defaults)
 
 set(CMAKE_MODULE_PATH ${cantor_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
 
+macro_optional_find_package(Analitza)
+macro_log_feature(ANALITZA_FOUND "libanalitza" "A library provided by KAlgebra" FALSE "" "Backend to use KAlgebra with Cantor.")
+
 macro_optional_find_package(R)
 macro_log_feature(R_FOUND "R" "A free software environment for statistical computing and graphics" "http://www.r-project.org/" FALSE "" "Backend to use R with Cantor.")
 
--- a/src/backends/CMakeLists.txt
+++ b/src/backends/CMakeLists.txt
@@ -2,9 +2,9 @@ add_subdirectory(null)
 add_subdirectory(maxima)
 add_subdirectory(octave)
 
-if(BUILD_kalgebra)
+if(ANALITZA_FOUND)
 	add_subdirectory(kalgebra)
-endif(BUILD_kalgebra)
+endif(ANALITZA_FOUND)
 
 if(NOT WIN32)
 	add_subdirectory(sage)
